src/main/kotlin/no/nav/syfo/oppfolgingstilfelle/person/domain/OppfolgingstilfelleDag.kt

```diff
@@ -76,36 +81,56 @@ private fun List<OppfolgingstilfelleDag>.findLastPriorityOppfolgingstilfelleBit(
         it.priorityOppfolgingstilfelleBit != null
     }.priorityOppfolgingstilfelleBit
 
-fun List<OppfolgingstilfelleDag>.gradertAtTilfelleEnd() =
-    this.findLastPriorityOppfolgingstilfelleBit()?.isGradert() ?: false
+private fun List<OppfolgingstilfelleDag>.findLastPriorityGraderingBit() =
+    this.lastOrNull { it.priorityGraderingBit != null }?.priorityGraderingBit
 
-fun List<OppfolgingstilfelleDag>.toOppfolgingstilfelle(): Oppfolgingstilfelle {
+fun List<OppfolgingstilfelleDag>.toOppfolgingstilfelle(
+    oppfolgingstilfelleDagerPerVirksomhet: Map<String?, List<OppfolgingstilfelleDag>>,
+): Oppfolgingstilfelle {
     if (this.onlySykmeldingNyOrInntektsmelding() && this.durationDays() > 118) {
         val sampleUUID = this.firstNotNullOfOrNull { dag -> dag.priorityOppfolgingstilfelleBit }?.uuid
         log.info("Created oppfolgingstilfelle with duration>118 days based on only sykmelding-ny, bit sample uuid: $sampleUUID")
         SYKMELDING_NY_COUNTER.increment()
     }
     val arbeidstakerAtTilfelleEnd = this.isArbeidstakerAtTilfelleEnd()
+    val tilfelleStart = this.first().dag
+    val tilfelleEnd = this.last().dag
+    val isGradertInAllVirksomheterAtTilfelleEnd =
+        gradertInAllVirksomheterAtTilfelleEnd(oppfolgingstilfelleDagerPerVirksomhet, tilfelleStart, tilfelleEnd)
+
     return Oppfolgingstilfelle(
         arbeidstakerAtTilfelleEnd = arbeidstakerAtTilfelleEnd,
-        start = this.first().dag,
-        end = this.last().dag,
+        start = tilfelleStart,
+        end = tilfelleEnd,
         virksomhetsnummerList = this.toVirksomhetsnummerPreferred().ifEmpty {
             if (arbeidstakerAtTilfelleEnd) this.toVirksomhetsnummerAll() else emptyList()
         },
-        gradertAtTilfelleEnd = this.gradertAtTilfelleEnd(),
+        gradertAtTilfelleEnd = isGradertInAllVirksomheterAtTilfelleEnd,
     )
 }
 
-private fun List<OppfolgingstilfelleDag>.onlySykmeldingNyOrInntektsmelding() =
-    this.all { dag ->
-        dag.priorityOppfolgingstilfelleBit?.tagList
-            ?.let { tagList ->
-                tagList in ((Tag.SYKMELDING and Tag.NY) or (Tag.INNTEKTSMELDING and Tag.ARBEIDSGIVERPERIODE))
-            }
-            ?: false
+private fun gradertInAllVirksomheterAtTilfelleEnd(
+    oppfolgingstilfelleDagerPerVirksomhet: Map<String?, List<OppfolgingstilfelleDag>>,
+    tilfelleStart: LocalDate,
+    tilfelleEnd: LocalDate,
+): Boolean {
+    val priorityGraderingBiter = oppfolgingstilfelleDagerPerVirksomhet.mapNotNull {
+        it.value.dagerInTilfelle(tilfelleStart, tilfelleEnd).findLastPriorityGraderingBit()
     }
 
+    return if (priorityGraderingBiter.all { it.isSykmeldingNy() }) priorityGraderingBiter.onlyGradert() else priorityGraderingBiter.filterNot { it.isSykmeldingNy() }
```

Comment:
Filtrerer ut sykmelding-ny-biter her siden graderings-informasjon fra andre biter skal overstyre disse.

Reply:
Gir det mening å flytte if-en inn i en egen funksjon med den informasjonen?

Reply:
Kan i hvert fall legge inn kommentar i koden - vanskelig å komme på noe godt navn på funksjon synes jeg :/
Kan evt skrive det som:
`return priorityGraderingBiter.filterNot { it.isSykmeldingNy() }.ifEmpty { priorityGraderingBiter }.onlyGradert()`
men usikker på om det er mer lesbart